src/engine/renderer/tr_shade.cpp

```diff
@@ -583,32 +583,35 @@ void Tess_Begin( void ( *stageIteratorFunc )(),
 SetNormalScale
 ==============
 */
-void SetNormalScale( shaderStage_t *pStage, vec3_t normalScale)
+void SetNormalScale( shaderStage_t *pStage, vec3_t normalScale )
 {
-		float normalIntensity = RB_EvalExpression( &pStage->normalIntensityExp, 1 );
-
-		normalScale[ 0 ] = 1;
-		normalScale[ 1 ] = 1;
-		normalScale[ 2 ] = 1;
-
-		normalScale[ 0 ] *= normalIntensity;
-		normalScale[ 1 ] *= normalIntensity;
+	float normalIntensity = RB_EvalExpression( &pStage->normalIntensityExp, 1.0 );
 
+	for ( int i = 0; i < 3; i++ )
+	{
 		if ( pStage->hasNormalScale )
 		{
-			normalScale[ 0 ] *= pStage->normalScale[ 0 ];
-			normalScale[ 1 ] *= pStage->normalScale[ 1 ];
-			normalScale[ 2 ] *= pStage->normalScale[ 2 ];
+			normalScale[ i ] = pStage->normalScale[ i ];
+		}
+		else
+		{
+			// assume OpenGL +X +Y +Z format if unknown
+			normalScale[ i ] = 1.0;
```

Comment:
So this is reversing the default from what it was before?

Reply:
There was no "default" in such meaning, neither a “-1” neither a “1” was stored. Parser was just feeding files to the renderer without knowing anything about the file. By introducing format management, this introduce a default format that is “1, 1 ,1”.

The default for “normalMap” keyword is  “1, -1, 1”.

In a way, we can say it reverse the default because if an existing normal map is loaded without the “normalMap” keyword (option we don't have anyway), it would be rendered wrong.

I decided to use “1, 1, 1” as a default because it makes things meaningful, things are written in a way they are, not in a way they must be written to be rendered right. The alternative way would be to name “1, -1, 1” OpenGL, and “1, 1, 1” DirectX, which would be confusing and bite us hard in the back in the future.

Reply:
We can say the default normal format for the material parser is OpenGL (something we did not have), but the default normal format in case of `normalMap` keyword is DirectX (like before).

Currently, this code is dead code, the xreal `normaMap` parser sets the DirectX format, the darkplaces parser sets the OpenGL format. We currently don't offer way to set normal maps without format being set. This code was thought to make it future proof but it's maybe a bad idea to fix potential issues silently, while the fix may be wrong.

I'm thinking about setting an UNREACHABLE assert there instead. This way we would ensure future implementation don't forget to set the format. But to do that we need a reliable way to know there is a normal map, and the code is not ready to do it easily (at this point we still have some materials storing normal map in diffuse slot). I'm not going to do the required refactorisation before that MR. This is planned, but this refactorization is not required for the release, while this normal format management is. I'll add a FIXME.

Reply:
Added `FIXME`s.